### PR TITLE
statistics: change read/write cache kinds to leader/peer

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What is changed and how it works?
Replace `ReadCache/WriteCache` with `LeaderCache/PeerCache` so it can reflect the intent better.
Old logic unchanged except for adding some more metrics.

### Check List

Tests
- Unit test

### Release note
- No release note

### CODE REVIEW

File: server/statistics/hot_peer_cache.go [hunk 1]
```diff
@@ -209,41 +209,50 @@ func (f *hotPeerCache) CheckRegionFlow(region *core.RegionInfo) (ret []*HotPeerS
...
+func (f *hotPeerCache) CollectMetrics() {
 	for storeID, peers := range f.peersOfStore {
 		store := storeTag(storeID)
 		thresholds := f.calcHotThresholds(storeID)
-		hotCacheStatusGauge.WithLabelValues("total_length", store, typ).Set(float64(peers.Len()))
-		hotCacheStatusGauge.WithLabelValues("byte-rate-threshold", store, typ).Set(thresholds[byteDim])
-		hotCacheStatusGauge.WithLabelValues("key-rate-threshold", store, typ).Set(thresholds[keyDim])
-		// for compatibility
-		hotCacheStatusGauge.WithLabelValues("hotThreshold", store, typ).Set(thresholds[byteDim])
+		hotCacheStatusGauge.WithLabelValues("total_length", store, f.kind.String()).Set(float64(peers.Len()))
```
Comment (rleungx) [inline server/statistics/hot_peer_cache.go:224]: I find the formats of labels are not consistent

Comment (disksing) [reply server/statistics/hot_peer_cache.go:224]: em, do you mean it should be `total-length`?

Comment (rleungx) [reply server/statistics/hot_peer_cache.go:224]: Yes. We have `byte-rate-threshold`, `hotThreshold`, and `total_length` three different kinds of format.

**Pull request closed without merging**